UMA-Bots/Funds-Deposited/src/agent.spec.ts

```diff
@@ -0,0 +1,165 @@
+import { Finding, HandleTransaction, ethers } from "forta-agent";
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { createAddress } from "forta-agent-tools";
+import { FUNDS_DEPOSITED_EVENT } from "./ABI";
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData } from "./config";
+import bot from "./agent";
+import { createFinding } from "./findings";
+import { BigNumber } from "ethers";
+
+describe("funds deposited bot", () => {
+  const MOCK_CONTRACT_ADDRESS = createAddress("0x1234");
+
+  let handleTransaction: HandleTransaction;
+
+  let mockTxEvent: TestTransactionEvent;
+
+  let networkManager: NetworkManager<NetworkData>;
+
+  let eventFragment: ethers.utils.EventFragment;
+  let mockEventFragment: ethers.utils.EventFragment;
+
+  beforeEach(() => {
+    //create test transaction before each test
+    mockTxEvent = new TestTransactionEvent();
+  });
+
+  beforeAll(() => {
+    const mockData: Record<number, NetworkData> = {
+      1111: {
+        spokePoolAddress: MOCK_CONTRACT_ADDRESS,
+      },
+    };
+
+    //initialize network manager with mock network
+    networkManager = new NetworkManager(mockData, 1111);
+
+    handleTransaction = bot.provideHandleTransaction(networkManager);
+
+    eventFragment = ethers.utils.EventFragment.from(FUNDS_DEPOSITED_EVENT.slice("event ".length));
+    mockEventFragment = ethers.utils.EventFragment.from("MockEvent(uint256)");
+  });
+
+  it("returns empty findings if no events are emitted,", async () => {
+    mockTxEvent.addEventLog(mockEventFragment, MOCK_CONTRACT_ADDRESS, [123]);
+
+    const findings: Finding[] = await handleTransaction(mockTxEvent);
+
+    expect(findings).toStrictEqual([]);
+  });
+
```

Comment:
here  `it("should ignore target events emitted from another contract")` and `it("should ignore other events emitted from target contract"` test cases would be nice to add.